crates/sui-core/src/epoch/reconfiguration.rs

```diff
@@ -52,7 +52,7 @@ where
             unreachable!();
         }
 
-        self.state.halted.store(true, Ordering::SeqCst);
+        self.state.halt_validator();
```

Comment:
I think that when we halt the validator not only we need to wait for all tickets to be drained, but also we need to wait for all transactions to be in batches, and also in checkpoint proposals, right?